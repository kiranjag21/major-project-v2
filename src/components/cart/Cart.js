import axios from 'axios';
import React from 'react';
import Deliveryaddress from './Deliveryaddress';
import CartDetails from './CartDetails';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

const mapStateToProps = state => {
	return {
		selectedDish: state.selectedDish.selectedDish
	}
}

class Cart extends React.Component{
  constructor(props){

    super(props);
    this.state={
      selectedDish: this.props.selectedDish,
      coupon:''
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(event){
    console.log(event.target.value);
    this.setState({
      coupon : event.target.value
    })
  }

  handleSubmit(event){
      event.preventDefault();
  }

  render(){
    console.log(this.state.selectedDish);
    if(this.state.selectedDish.dishes){
      return(
        <div className="container justify-content-center mt-3 bg-light  h-100" style={{ fontFamily: 'poppins'}}>
          <div className="row justify-content-center border-left">
            {/* <marquee><h5>Apply coupon <span className="text-success">FIRST</span> to get 10% discount on your first order! </h5></marquee> */}
          </div>
          <div className="row justify-content-center">
          <div className="col-8 mr-md-5 mt-1 ">

                <CartDetails details={this.state.selectedDish} coupon={this.state.coupon}></CartDetails>

            </div>

            <div className="col mt-5  ">
              < Deliveryaddress data={this.state.selectedDish}/>
              <div className="card border border-dark mt-5">
              <div className="card-body">
                 <h5 className="card-title float-left"><h5>Apply coupon <span className="text-success">FIRST</span> to get 10% discount on your order! </h5> </h5><br/>
              </div>
            </div>
              <form className="mt-5">
                <input type="text" name="name" value={this.state.coupon} onChange={this.handleChange}/><br/>
                <button type="button" class="btn btn-success"> Apply </button>
              </form>
            </div>

          </div>
      </div>
      )
    }else{
      return(
        <div className="container justify-content-center mt-3 bg-light " style={{fontFamily:'Poppins, sans-serif'}}>
          <div className="row justify-content-center border-left">
            {/* <marquee><h5>Apply coupon <span className="text-success">FIRST</span> to get 10% discount on your first order! </h5></marquee> */}
          </div>
          <div className="row justify-content-center">
          <div className="col mr-md-5 mt-1 ">

                <CartDetails details={this.state.selectedDish} coupon={this.state.coupon}></CartDetails>

            </div>
          </div>
      </div>
      )
    }


}

}
export default withRouter(connect(mapStateToProps)(Cart));