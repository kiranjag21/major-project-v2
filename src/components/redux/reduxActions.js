export const addItems = (selectedDish) => ({
  type: 'ADD_ITEMS',
  payload: selectedDish
});

export const changeQuantity = (id,quantity)=>({
type: 'CHANGE_QUANTITY',
payload : {
  id:id,
  quantity:quantity
}
});

export const removeItem = (id)=>({
type:'REMOVE_ITEM',
payload:id
});

export const clearCart =()=>({
type:'CLEAR_CART',

})
export const addUser = (id) => ({
  type: 'ADD_USER',
  payload: id
});
export const removeUser = () => ({
  type: 'REMOVE_USER'
});
export const setConnection = (con) => ({

  type: 'ADD_CONNECTION',
  payload: con
});

export const disconnect = () => ({

  type: 'DISCONNECT'
});