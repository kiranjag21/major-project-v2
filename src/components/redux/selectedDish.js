
export const SelectedDish = (state = {
  selectedDish: []
}, action) => {
// let selectedDish = state.selectedDish;
  switch (action.type) {

      case 'ADD_ITEMS':
          return { ...state, selectedDish: action.payload };

      case 'CHANGE_QUANTITY':
         console.log(state.selectedDish.dishes);
        let item = state.selectedDish.dishes.find((item) => item.dishId === action.payload.id);
        let newCart = state.selectedDish.dishes.filter((item) => item.dishId !== action.payload.id);
        item.quantity = action.payload.quantity;

        newCart.push(item);
        state.selectedDish.dishes=newCart;
        // console.log(newCart);
        return{...state,selectedDish:state.selectedDish};

         case 'REMOVE_ITEM':
           console.log("aaya");
           console.log(state.selectedDish.dishes);
           console.log(action.payload.id);
           let newCart1 = state.selectedDish.dishes.filter((item) => item.dishId !== action.payload);
            console.log(newCart1);
            state.selectedDish.dishes=newCart1;
            console.log(newCart1);
            return{...state,selectedDish:state.selectedDish};

         case 'CLEAR_CART':
          let selectedDish=[];
         return{...state,selectedDish:selectedDish};

      default:
          return state;
  }
}