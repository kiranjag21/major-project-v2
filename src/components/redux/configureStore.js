
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { Connection } from './pusherConnection';
import { SelectedDish } from './selectedDish';
import { Auth } from './auth';
export const configureStore = () => {
    const store = createStore(
        combineReducers({
            selectedDish: SelectedDish,
            connection: Connection,
            auth: Auth
        }),
        applyMiddleware(thunk, logger)
    );

    return store;
}