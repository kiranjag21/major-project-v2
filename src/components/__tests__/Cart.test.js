import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import { Provider } from 'react-redux';
import { configureStore } from '../redux/configureStore';
import Cart from './../cart/Cart';

const store=configureStore();
test('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Provider store={store}><Cart/></Provider></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
