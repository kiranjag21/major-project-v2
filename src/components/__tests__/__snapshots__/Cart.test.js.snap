// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container justify-content-center mt-3 bg-light "
  style={
    Object {
      "fontFamily": "Poppins, sans-serif",
    }
  }
>
  <div
    className="row justify-content-center border-left"
  />
  <div
    className="row justify-content-center"
  >
    <div
      className="col mr-md-5 mt-1 "
    >
      <div
        className="empty-cart"
      >
        <img
          alt="empty-cart"
          src="https://res.cloudinary.com/sivadass/image/upload/v1495427934/icons/empty-cart.png"
        />
        <h2
          className="text-warning text-center"
        >
          You cart is empty!
        </h2>
      </div>
    </div>
  </div>
</div>
`;
